libs/block-editor/src/lib/extensions/dragHandler.extention.ts

```diff
@@ -0,0 +1,158 @@
+import { Extension } from '@tiptap/core';
+import { NodeSelection, Plugin, PluginKey } from 'prosemirror-state';
+import { serializeForClipboard } from 'prosemirror-view/src/clipboard';
+import { EditorView } from 'prosemirror-view';
+
+export const DragHandler = Extension.create({
+    name: 'dragHandler',
+
+    addProseMirrorPlugins() {
+        const brokenClipboardAPI = false; // (browser.ie && browser.ie_version < 15) || (browser.ios && browser.webkit_version < 604)
+
+        function createRect(rect) {
+            if (rect == null) {
+                return null;
+            }
+            let newRect = {
+                left: rect.left + document.body.scrollLeft,
+                top: rect.top + document.body.scrollTop,
+                width: rect.width,
+                height: rect.height,
+                bottom: 0,
+                right: 0
+            };
+            newRect.bottom = newRect.top + newRect.height;
+            newRect.right = newRect.left + newRect.width;
+            return newRect;
+        }
+
+        function absoluteRect(element: HTMLElement): ClientRect {
+            return createRect(element.getBoundingClientRect());
+        }
+
+        function removeNode(node) {
+            if (node && node.parentNode) {
+                node.parentNode.removeChild(node);
+            }
+        }
+
+        function blockPosAtCoords(coords, view) {
+            let pos = view.posAtCoords(coords);
+            let node = view.domAtPos(pos.pos);
+
+            node = node.node;
+            while (node && node.parentNode) {
+                if (node.parentNode.classList.contains('ProseMirror')) {
+                    // todo
+                    break;
+                }
+                node = node.parentNode;
+            }
+            if (node && node.nodeType === 1) {
+                let desc = view.docView.nearestDesc(node, true);
+                if (!(!desc || desc === view.docView)) {
+                    return desc.posBefore;
+                }
+            }
+            return null;
+        }
+
+        function dragStart(e, view: EditorView) {
+            if (!e.dataTransfer) return;
+
+            let coords = { left: e.clientX + 50, top: e.clientY };
+            let pos = blockPosAtCoords(coords, view);
+            if (pos != null) {
+                view.dispatch(
+                    view.state.tr.setSelection(NodeSelection.create(view.state.doc, pos))
+                );
+
+                let slice = view.state.selection.content();
+                let { dom, text } = serializeForClipboard(view, slice);
+
+                e.dataTransfer.clearData();
+                e.dataTransfer.setData(brokenClipboardAPI ? 'Text' : 'text/html', dom.innerHTML);
+                if (!brokenClipboardAPI) e.dataTransfer.setData('text/plain', text);
+
+                view.dragging = { slice, move: true };
```

Comment:
set the dragging content to the editor, then the editor will handle the drop "automatically".